controllers/pod_controller.go

```diff
@@ -77,18 +69,17 @@ func (r *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		matched[s] = ""
 	})
 
-	shouldRetry := ctrl.Result{RequeueAfter: 10 * time.Second}
+	shouldRetry := ctrl.Result{RequeueAfter: 10 * time.Minute}
 
 	admitResult := admitPod(&pod)
 	switch admitResult {
 	case ValidationStatusSuccess:
-		l.Info("pod validated without errors")
+		l.Info("pod validated is successful")
 		shouldRetry = ctrl.Result{}
 		break
 	case ValidationStatusFailed:
 		//TODO this should return some kind of error
-		l.Info("pod validated with errors")
-		shouldRetry = ctrl.Result{}
+		l.Info("pod validated failed")
 		break
 	}
 	pod.Labels[PodValidationLabel] = admitResult
```

Comment:
If pod admission result is the same as the current label value then there is no need to update the resource. I think the entire update function should be also covered by a logic, where it runs the update only if `pod.Labels[PodValidationLabel] != admitResult`

Reply:
actually, no update is applied in ETC if the content of the resource doesn't change, otherwise setting `failed` would lead to indefinite reconciliation. However, I put it as a defensive check.

Reply:
Oh, it's good to know that. Guess I will have to learn more about all these internal caveats 👍